### PR TITLE
Clarify when the shared secret is needed

### PR DESCRIPTION
**Pull request by Ordinant**
## Description

Clarify that setting a shared secret is no longer optional for any authentication, and that configuring internal TLS communication is a separate and optional feature. 

> Is this change a fix, improvement, new feature, refactoring, or other?
Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)
Docs only

> How would you describe this change to a non-technical end user or system administrator?
Doc clarification

## Related issues, pull requests, and links

<!-- List any issues fixed by this PR, and provide links to other related PRs, upstream release notes, and other useful resources. For example:
* Fixes #issuenumber
* Related documentation in #issuenumber
* [Some release notes](http://usefulinfo.example.com)

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/security/internal-communication.rst [hunk 1]
```diff
@@ -2,24 +2,29 @@
 Secure internal communication
 =============================
 
-The Trino cluster can be configured to use secured communication with internal
-authentication of the nodes in the cluster, and optionally added security with
-:ref:`TLS <glossTLS>`.
+You must configure your Trino cluster to use a shared secret to secure
```
Comment (mosabua) [inline docs/src/main/sphinx/security/internal-communication.rst:5]: That implies that is is required.. which is not true.. only required if you want to use any authentication or if you want internal TLS

**Review by mosabua — CHANGES_REQUESTED** [review]
Update as with comment and either get rid of unrelated changes or separate into two commits

File: docs/src/main/sphinx/security/internal-communication.rst (hunk 1 above)
Comment (Ordinant) [reply docs/src/main/sphinx/security/internal-communication.rst:5]: "Absolutely required" is what I kept asking you, and you kept saying yes ...

Comment (mosabua) [reply docs/src/main/sphinx/security/internal-communication.rst:5]: well.. it is not required if you have no authentication and do not want to use internal TLS... thats what I tried to communicate .. essentially its required in nearly all real world usage scenarios..

Comment (Ordinant) [reply docs/src/main/sphinx/security/internal-communication.rst:5]: ✅

**Review by mosabua — APPROVED** [review]
Great .. let's ship this.

**Comment by Ordinant** [comment]
Further clarification suggested by David is in place now. Lines 12-13

**Review by mosabua — APPROVED** [review]

**Comment by mosabua** [comment]
Can we get this merged now @electrum ?

**Comment by mosabua** [comment]
@electrum @martint .. can we get this merged..

**Pull request merged**